tests/test_quixstreams/test_sinks/test_core/test_influxdb_v3.py

```diff
@@ -363,3 +365,47 @@ def test_write_fails_propagates_exception(self, influxdb3_sink_factory):
         )
         with pytest.raises(influxdb_client_3.InfluxDBError):
             sink.flush()
+
+    @pytest.mark.parametrize(
+        "fields_keys, result",
+        [
+            ((), {"str_key": "value", "int_key": 0.0, "float_key": 1.1}),
+            (("str_key", "int_key"), {"str_key": "value", "int_key": 0.0}),
+        ],
+    )
+    def test_convert_ints_to_floats(self, influxdb3_sink_factory, fields_keys, result):
+        client_mock = MagicMock(spec_set=InfluxDBClient3)
+        measurement = "measurement"
+        sink = influxdb3_sink_factory(
+            client_mock=client_mock,
+            measurement=measurement,
+            fields_keys=fields_keys,
+            convert_ints_to_floats=True,
+        )
+        topic = "test-topic"
+
+        value, timestamp = {"str_key": "value", "int_key": 0, "float_key": 1.1}, 1
+        sink.add(
+            value=value,
+            key="key",
+            timestamp=timestamp,
+            headers=[],
+            topic=topic,
+            partition=0,
+            offset=1,
+        )
+        sink.flush()
+
+        assert client_mock.write.call_count == 1
+        first_call = client_mock.write.call_args_list[0]
+        assert first_call.kwargs == {
+            "record": [
+                {
+                    "measurement": measurement,
+                    "tags": {},
+                    "fields": result,
+                    "time": timestamp,
+                }
+            ],
+            "write_precision": "ms",
+        }
```

Comment:
There is a shorthand for this
```suggestion
        client_mock.write.assert_called_once_with(
            record=[
                {
                    "measurement": measurement,
                    "tags": {},
                    "fields": result,
                    "time": timestamp,
                }
            ],
            write_precision="ms",
        )
```

Reply:
Indeed, I was mostly just copying the previous test patterns for consistency, but this is obviously the better way to do it =)

I can also tweak the other ones to match this style if you think it's worthwhile, or just leave it all as-is.

Reply:
up to you man 😎
it's not a big deal, I'd probably do it